### ISSUE TITLE
getAnimeSearch v4 API returns inconsistent datatype for different pages

### ISSUE THREAD

**Issue by aebeljs** [jikan-me/jikan-rest#298]
`getAnimeSearch` API returns an array in the `data` field when the query parameter `page` is given as `1`. However the same API returns an object in the `data` field when `page` is given as anything other than `1`. Can we have the same format in all cases?

When `page` is `1`
![image](https://user-images.githubusercontent.com/29663106/204223152-ec18af10-e81a-4718-afe5-77e3ea9377d7.png)

When `page` is not `1`
![image](https://user-images.githubusercontent.com/29663106/204223294-024be6a0-3de4-4406-a862-a650f39b45e2.png)

**Comment by irfan-dahir** [comment]
Looks like this only happens when there's only 1 result

**Comment by pushrbx** [comment]
I'm thinking that the mixin for the scout builder I added causes this issue. Also I've found one more bug: when you ask for the first page, it only returns the total number of items in the current page, not in overall if Typesense is enabled.

```
https://api.jikan.moe/v4/anime?q=ae&page=1
```
`total_count: 27`

I think the scout driver for Typesense is not using the correct field from the typesense api response. It should use the `out_of` field.

https://typesense.org/docs/0.23.1/api/search.html#search
